### PR TITLE
Add auth middleware and service features

### PR DESCRIPTION
**Pull request by allthesignals**
This commit exposes a ‘session’ variable in the request object.

### CODE REVIEW

File: server/src/auth/auth.service.ts [hunk 1]
```diff
@@ -17,8 +17,8 @@ import { ContactService } from '../contact/contact.service';
 @Injectable()
```
Comment (godfreyyeung) [inline server/src/auth/auth.service.ts:17]: Is this file copy and pasted from old zap-api? There are a lot of documentation, code clarity and logging improvements in the current authService file.

File: server/src/auth/auth.service.ts [hunk 2]
```diff
@@ -27,34 +27,77 @@ export class AuthService {
...
+   * how authentication actually works.
+   * 
+   * Once the cookie is printed in the terminal, you can copy that cookie and paste
+   * it into headers of requests you're making. If you're looking at the API and
+   * want it to persist, you can use the Chrome Inspector to add the token via the
+   * Application tab.
+   * 
+   * See this PR for full discussion: https://github.com/NYCPlanning/zap-api/pull/74#discussion_r370749677.
+   * 
+   * @param      {<type>}  NYCIDToken  The nycid token
```
Comment (godfreyyeung) [inline server/src/auth/auth.service.ts:60]: return type should be {string} here?

File: server/src/auth/auth.service.ts [hunk 3]
```diff
@@ -17,8 +17,8 @@ import { ContactService } from '../contact/contact.service';
 @Injectable()
 export class AuthService {
   // required env variables
-  NYCID_TOKEN_SECRET = '';
-  ZAP_TOKEN_SECRET = '';
+  CRM_SIGNING_SECRET = '';
+  NYCID_CONSOLE_PASSWORD = '';
```
[the remark this answers is not in the document]
Comment (allthesignals) [reply server/src/auth/auth.service.ts:21]: yeah i just copied it, i'l change it back

[the remark this answers is not in the document]
Comment (godfreyyeung) [reply server/src/auth/auth.service.ts:21]: Okay, Thanks!

**Pull request merged**